packages/glimmer-runtime/lib/environment.ts

```diff
@@ -200,21 +200,31 @@ export abstract class Environment {
 
 export default Environment;
 
-// TS does not allow us to use computed properties for this, so inlining for now
-// import { TRUSTED_STRING } from './symbols';
+export interface SafeString {
+  toHTML(): string;
+}
+
+export function isSafeString(value: Opaque): value is SafeString {
+  return value && typeof value['toHTML'] === 'function';
+}
+
+export function isNode(value: Opaque): value is Node {
+  return value !== null && typeof value === 'object' && typeof value['nodeType'] === 'number';
+}
 
-interface SafeString {
-  "trusted string [id=7d10c13d-cdf5-45f4-8859-b09ce16517c2]": boolean; // true
-  string: string;
+export function isString(value: Opaque): value is string {
+  return typeof value === 'string';
 }
 
-export type Insertion = string | SafeString | Node;
+export type Insertion = CautiousInsertion | TrustingInsertion;
+export type CautiousInsertion = string | SafeString | Node;
+export type TrustingInsertion = string | Node;
 
 type PositionalArguments = Opaque[];
 type KeywordArguments = Dict<Opaque>;
 
 export interface Helper {
-  (args: EvaluatedArgs): PathReference<Opaque>;
```

Comment:
I was surprised that this change didn't cause any type errors, and I just figured out why. In https://github.com/tildeio/glimmer/blob/master/packages/glimmer-runtime/lib/compiled/expressions/helper.ts#L22, we used `Function.prototype.call` which is typed to be `call(thisArg: any, ...args: any[]): any`, so the return value became `any` and the type system happily let it go through, even though the method is supposed to return a `PathReference<Opaque>` 😞

Fixing that as well